Проверьте Курсовую работу №1

Tatyana58 opened the pull request.

Tatyana58 committed:
Начало
Создали класс и конструктор Employee

Tatyana58 committed:
Начало
Создали класс и конструктор Employee

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.
Поиск сотрудника с максимальной зарплатой

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.
Поиск сотрудника с максимальной зарплатой

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.
Поиск сотрудника с максимальной зарплатой
Поиск сотрудника с максимальной зарплатой в отделе

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.
Поиск сотрудника с максимальной зарплатой
Поиск сотрудника с максимальной зарплатой в отделе
Всех сотрудников с зарплатой меньше числа

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.
Поиск сотрудника с максимальной зарплатой
Поиск сотрудника с максимальной зарплатой в отделе
Всех сотрудников с зарплатой меньше числа
Поиск всех сотрудников с зарплатой больше (или равно) числа

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.
Поиск сотрудника с максимальной зарплатой
Поиск сотрудника с максимальной зарплатой в отделе
Всех сотрудников с зарплатой меньше числа
Поиск всех сотрудников с зарплатой больше (или равно) числа
Курсовая работа выполнена **Повышенная сложность

zrazzhivin commented on src/Main.java:
```diff
@@ -1,5 +1,304 @@
+
+    //Поиск среднего значения зарплат:
+    public static double averageSumWages(Employee[] collegue) {
+        sumAll=sumAll/ collegue.length;
```
тут правильнее было бы обратиться к методу , который подсчитывает сумму. У нас нет гарантии, что свойство sumAll что-то содержит. не нужно на это надеяться и просто вызываем метод, который считает сумму.

zrazzhivin commented on src/Main.java:
```diff
@@ -1,5 +1,304 @@
+        for (int i = 0; i < collegue.length; i++) {
+            System.out.println("" + collegue[i].getSurname().toUpperCase() + "" + collegue[i].getName().toUpperCase() + "" + collegue[i].getPatronymic().toUpperCase());
+        }
+        return "";
```
если не хотим ничего возвращать - тогда используем в сигнатуре метода void. Возвращать пустую строку - не совсем очевидное решение.

Tatyana58 committed:
Начало
Создали класс и конструктор Employee
Переопределили метод toString equals hashCode
Посчитали сумму затрат на зарплаты в месяц
Список всех сотрудников через toString
Сотрудник с минимальной зарплатой.
Сотрудник с максимальной зарплатой.
Среднее значение зарплат.
Список Ф.И.О сотрудников
Список Ф.И.О сотрудников в верхнем регистре
Замена номера отдела
Изменение заработной платы по id
Поиск сотрудника по отделу
Поиск сотрудника по id
Индексация заработной платы в %
Изменила Поиск сотрудников по отделу
Изменен поиск сотрудника с максимальной и с минимальной зарплатой.
Поиск суммы затрат на зарплаты в месяц по отделу.
Среднюю зарплату по отделу
Проиндексировать зарплату всех сотрудников отдела на процент
Поиск сотрудника с минимальной зарплатой в отделе.
Поиск сотрудника с максимальной зарплатой
Поиск сотрудника с максимальной зарплатой в отделе
Всех сотрудников с зарплатой меньше числа
Поиск всех сотрудников с зарплатой больше (или равно) числа
Курсовая работа выполнена **Повышенная сложность
Изменила вывод данных.- убрала return ""
Изменила Поиск среднего значения зарплат:

Tatyana58 commented:
Изменила Курсовую работу
